### PR TITLE
#1906 enforce indicator types

### LINKED ISSUE #1906: Enforce indicator types
## Is your feature request related to a problem? Please describe.

Indicator value updates are currently untyped.

## Describe the solution you'd like

Associated `[indicator_attribute]value_type` should be used for enforcing type constraints.

## Implementation

Update `editIndicatorValue` cell action.

## Describe alternatives you've considered

N/A.

## Additional context

N/A.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #1906.

## Change summary

Adds type enforcement to indicator data entry.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] HIV input cells can only be set to positive integer values.
- [ ] HIV input cell values are persisted after reopening requisition.
- [ ] Requisition value input cells can only be set to the row data type.
- [ ] Requisition comment input cells can be any string.

### Related areas to think about

Currently, indicators only supports string and numerical (interpreted as positive integer) data. Would be good in future to add support for negative integers, real numbers, booleans etc.

### CODE REVIEW

File: src/sync/incomingSyncUtils.js [hunk 1]
```diff
@@ -337,7 +336,7 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
         period: database.getOrCreate('Period', record.period_ID),
         column: indicatorColumn,
         row: indicatorRow,
-        value: record.value ?? '',
+        _value: record.value ?? '',
```
Comment (joshxg) [inline src/sync/incomingSyncUtils.js]: should this default to `"0"` for numeric types

Comment (wlthomson) [reply src/sync/incomingSyncUtils.js]: Currently all initialisation for `IndicatorValue` records is handled on creation, where the initial value can inferred from the column, row types). When the value is being synced in, there's no guarantee the `indicatorColumn` or `indicatorRow` has been synced yet (although likely it has), but also if that record has a `null` or `undefined` value,  it means somethings gone wrong big time (as it should never be able to get to that state).

**Comment by wlthomson** [comment]
> Request changes to check if this be into a patch? And therefore master?

Good point. What do you think about making a patch release branch off master?

**Comment by joshxg** [comment]
sure sounds good

**Pull request merged**